tests/components/FilterSearch.test.tsx

```diff
@@ -0,0 +1,298 @@
+    });
+
+    jest.spyOn(searchOperations, 'executeSearch').mockImplementation();
+  });
+
+  it('renders the filter search bar and the "Filter" label', () => {
+    render(<FilterSearch searchFields={searchFieldsProp}/>);
+
+    const label = 'Filter';
+    const labelElement = screen.getByText(label);
+    const searchBarElement = screen.getAllByRole('textbox');
+
+    expect(labelElement).toBeDefined();
+    expect(searchBarElement.length).toBe(1);
+  });
+
+  it('displays characters typed in search bar correctly',() => {
+    render(<FilterSearch searchFields={searchFieldsProp}/>);
+
+    const searchBarElement = screen.getByRole('textbox');
+
+    userEvent.type(searchBarElement, 'n');
+    expect(searchBarElement).toHaveValue('n');
+
+    userEvent.type(searchBarElement, 'a');
+    expect(searchBarElement).toHaveValue('na');
+
+    userEvent.type(searchBarElement, '{backspace}');
+    expect(searchBarElement).toHaveValue('n');
+  });
+
+  it('triggers a filter search every time a character is typed or backspaced', () => {
+    render(<FilterSearch searchFields={searchFieldsProp}/>);
+
+    const searchBarElement = screen.getByRole('textbox');
+
+    userEvent.type(searchBarElement, 'n');
+    expect(actions.executeFilterSearch).toHaveBeenLastCalledWith('n', false, expectedFilterSearchFields);
+
+    userEvent.type(searchBarElement, 'a');
+    expect(actions.executeFilterSearch).toHaveBeenLastCalledWith('na', false, expectedFilterSearchFields);
+
+    userEvent.type(searchBarElement, '{backspace}');
+    expect(actions.executeFilterSearch).toHaveBeenLastCalledWith('n', false, expectedFilterSearchFields);
+
+    expect(actions.executeFilterSearch).toHaveBeenCalledTimes(3);
+  });
+
+  it('does not trigger a filter search when backspacing in an empty text box', () => {
+    render(<FilterSearch searchFields={searchFieldsProp}/>);
+
+    const searchBarElement = screen.getByRole('textbox');
+
+    userEvent.type(searchBarElement, '{backspace}');
+    expect(searchBarElement).toHaveValue('');
+
+    expect(actions.executeFilterSearch).toHaveBeenCalledTimes(0);
+  });
+
+  it('shows autocomplete results, if they exist, when a character is typed', async () => {
+    render (<FilterSearch searchFields={searchFieldsProp}/>);
+
+    const searchBarElement = screen.getByRole('textbox');
+    userEvent.type(searchBarElement, 'n');
+
+    await waitFor(() => {
+      const firstAutocompleteSection= screen.getByText('First name');
+      expect(firstAutocompleteSection).toBeDefined();
+    });
+
+    await waitFor(() => {
+      const firstAutocompleteSuggestion = screen.getByText('first name 1');
+      expect(firstAutocompleteSuggestion).toBeDefined();
+    });
+
+    await waitFor(() => {
+      const secondAutocompleteSuggestion = screen.getByText('first name 2');
+      expect(secondAutocompleteSuggestion).toBeDefined();
+    });
+
+    await waitFor(() => {
+      const secondAutocompleteSection= screen.getByText('Last name');
```

Comment:
does eslint point out the missing space before equals? If not we can update our common slapshot eslint sometime later